config/v1/types_infrastructure.go

```diff
@@ -189,6 +192,45 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+type CloudControllerManagerState string
+
+const (
+	// Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling that kubelets and other CCM consumers should use --cloud-provider=external.
```

Comment:
Given that we know that `--cloud-provider=external` is going away, and in the future we will have to manually taint the new nodes, perhaps we want to reword this.

I'm thinking something like

```
// Cloud Controller Manager is enabled and expected to be installed by a third party operator.
// This field indicates to the platform that new nodes should be tainted as uninitialized when created,
// preventing them from running workloads until they are initialized by the cloud controller manager.
```

Reply:
Updated. I just removed the statement about `a third party operator `, I don't think that it would be an operator necessarily.